File: epi_judge_python/count_bits.py
from test_framework import generic_test


def count_bits(x: int) -> int:
    bits = 0
    while x:
        x = x & (x - 1)
        bits += 1

    return bits


if __name__ == '__main__':
    exit(
        generic_test.generic_test_main('count_bits.py', 'count_bits.tsv',
                                       count_bits))

File: epi_judge_java/epi/CountBits.java
package epi;

import epi.test_framework.EpiTest;
import epi.test_framework.GenericTest;

public class CountBits {
  @EpiTest(testDataFile = "count_bits.tsv")

  public static short countBits(int x) {
    short bits = 0;

    while (x != 0) {
      x = x & (x - 1);
      bits++;
    }

    return bits;
  }

  public static void main(String[] args) {
    System.exit(
        GenericTest
            .runFromAnnotations(args, "CountBits.java",
                new Object() {
                }.getClass().getEnclosingClass())
            .ordinal());
  }
}
